jgrapht-io/src/main/java/org/jgrapht/io/DOTExporter.java

```diff
@@ -271,12 +273,30 @@ private void renderAttributes(PrintWriter out, String labelName, Map<String, Att
                     // already handled by special case above
                     continue;
                 }
-                out.print(name + "=\"" + entry.getValue().getValue() + "\" ");
+                out.print(name + "=" + quoteIfNeeded(escapeDoubleQuotes(entry.getValue().getValue())) + " ");
             }
         }
         out.print("]");
     }
 
+    private static String quoteIfNeeded(String id){
+        if (KEYWORDS.contains(id)) {
+            return "\"" + id + "\"";
+        } else {
+            return id;
+        }
+    }
+
+    private static String escapeDoubleQuotes(String id) {
```

Comment:
We depend on the Apache Commons Lang v3 library which contains a lot of stuff about escaping and unescaping. See for example the DOTImporter class.